src/path_rewriting.rs

```diff
@@ -709,8 +725,8 @@ mod tests {
         for (abs_path, rel_path, result) in results {
             count += 1;
             assert!(abs_path.is_absolute());
-            assert!(abs_path.ends_with("tests\\class\\main.cpp"));
-            assert_eq!(rel_path, PathBuf::from("tests\\class\\main.cpp"));
+            assert!(abs_path.ends_with("tests\\java\\main.java"));
+            assert_eq!(rel_path, PathBuf::from("tests\\java\\main.java"));
             assert_eq!(result, empty_result!());
         }
         assert_eq!(count, 1);
```

Comment:
Could you also add a test where the partial path is mapped for a java file and not for a cpp or rust file?